test/screenshot/text-field/fullWidth.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import {iconsMap, denseMap, rtlMap, requiredMap, disabledMap, helperTextMap} from './attributesMap';
+import TestField from './TestTextField';
+
+const getHelperKeyText = (helperText, helperTextIndex) => {
+  const hasHelperText = Object.keys(helperText).length > 0;
+  const isValidationMsg = helperTextIndex === 2;
+
+  if (!hasHelperText) {
+    return '';
+  } else if (isValidationMsg) {
+    return 'validation';
+  } else {
+    return 'persistent';
+  }
+};
+
+const textFields = iconsMap.map((icon) => {
```

Comment:
Hmm this is a lot of redundant code for every variant. Can we instead create a new component called `TextFieldPermutations` to dump all this code in? It can go in the TestTextField file, so it'd look like
```
class TestField extends React.Component {
  ...
}

class TextFieldPermutations extends React.Component {
  getHelperKeyText = (helperText) => {
    ...
  }

  render() {
    return (iconsMap.map((icon) => {
      return denseMap.map((dense) => {
        ...
        const props = Object.assign({}, this.props, icon, ...);
        ...
        return <TestField {...props} key={key} id={key} />;
      });
    );
  }
}
```

And then in this file we'd just have
```
const TextFieldScreenshotTest = () => (
  <div className='text-field-container'>
    <TextFieldPermutations fullWidth /  >
  </div>
);
```